src/IIIFPresentation/API.Tests/Integration/ModifyCollectionTests.cs

```diff
@@ -0,0 +1,319 @@
+            Slug = "programmatic-child",
+            Parent = parent
+        };
+
+        var requestMessage = new HttpRequestMessage(HttpMethod.Post, $"{Customer}/collections");
+        requestMessage.AddPrivateHeaders();
+        requestMessage.Content = new StringContent(JsonSerializer.Serialize(collection), Encoding.UTF8,
+            new MediaTypeHeaderValue("application/json"));
+        
+        // Act
+        var response = await httpClient.AsCustomer(1).SendAsync(requestMessage);
+
+        var responseCollection = await response.ReadAsPresentationResponseAsync<FlatCollection>();
+
+        var fromDatabase = dbContext.Collections.First(c => c.Id == responseCollection!.Id!.Split('/', StringSplitOptions.TrimEntries).Last());
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.Created);
+        fromDatabase.Parent.Should().Be(parent);
+        fromDatabase.Label.Values.First()[0].Should().Be("test collection");
+        fromDatabase.Slug.Should().Be("programmatic-child");
+        fromDatabase.IsPublic.Should().BeTrue();
+        fromDatabase.IsStorageCollection.Should().BeTrue();
+    }
+    
+    [Fact]
+    public async Task CreateCollection_FailsToCreateCollection_WhenDuplicateSlug()
+    {
+        // Arrange
+        var collection = new FlatCollection()
+        {
+            Behavior = new List<string>()
+            {
+                Behavior.IsPublic,
+                Behavior.IsStorageCollection
+            },
+            Label = new LanguageMap("en", ["test collection"]),
+            Slug = "first-child",
+            Parent = parent
+        };
+        
+        var requestMessage = new HttpRequestMessage(HttpMethod.Post, $"{Customer}/collections");
+        requestMessage.AddPrivateHeaders();
+        requestMessage.Content = new StringContent(JsonSerializer.Serialize(collection), Encoding.UTF8,
+            new MediaTypeHeaderValue("application/json"));
+
+        // Act
+        var response = await httpClient.AsCustomer(1).SendAsync(requestMessage);
+        
+        var error = await response.ReadAsPresentationResponseAsync<Error>();
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.BadRequest); 
+        error!.Detail.Should().Be("The collection could not be created due to a duplicate slug value");
+    }
+    
+    [Fact]
+    public async Task CreateCollection_FailsToCreateCollection_WhenCalledWithoutAuth()
+    {
+        // Arrange
+        var collection = new FlatCollection()
+        {
+            Behavior = new List<string>()
+            {
+                Behavior.IsPublic,
+                Behavior.IsStorageCollection
+            },
+            Label = new LanguageMap("en", ["test collection"]),
+            Slug = "first-child",
+            Parent = parent
+        };
+        
+        var requestMessage = new HttpRequestMessage(HttpMethod.Post, $"{Customer}/collections");
```

Comment:
Variations of these 3 lines are repeated a lot, could a test helper be used?

```cs
var requestMessage = new HttpRequestMessage(HttpMethod.Post, $"{Customer}/collections");
requestMessage.Headers.Add("IIIF-CS-Show-Extra", "All");
requestMessage.Content = new StringContent(JsonSerializer.Serialize(collection), Encoding.UTF8,
            new MediaTypeHeaderValue("application/json"));
```
could be

```cs
public static HttpRequestMessage GetPrivateRequest(HttpMethod method, string path, object content);
```